ccdproc/tests/test_cosmicray.py

```diff
@@ -21,17 +26,22 @@
 
 def add_cosmicrays(data, scale, threshold, ncrays=NCRAYS):
     size = data.shape[0]
-    rng = np.random.default_rng(99)
-    crrays = rng.integers(0, size, size=(ncrays, 2))
+    rng = default_rng(99)
+    crrays = xp.asarray(rng.integers(0, size, size=(ncrays, 2)))
     # use (threshold + 15) below to make sure cosmic ray is well above the
     # threshold no matter what the random number generator returns
     # add_cosmicrays is highly sensitive to the seed
     # ideally threshold should be set so it is not sensitive to seed, but
     # this is not working right now
-    crflux = 10 * scale * rng.random(NCRAYS) + (threshold + 15) * scale
+    crflux = xp.asarray(10 * scale * rng.random(ncrays) + (threshold + 15) * scale)
+
+    # Some array libraries (Dask) do not support setting individual elements,
```

Comment:
Does `array_api_extras` address this? I haven't looked into the details

Reply:
It turns out the answer is no, but not for the reason the comment suggests. Using `xpx.at(...` raises a exception with the message `NotImplementedError: dask does not yet implement assignment to a scalar numpy or dask array index`. I think `array_api_extra` must convert the index to an array, and dask doesn't support indexing by an array.

Just using regular indexing (i.e. not `xpx.at`) works for everything except `jax`, where the arrays are immutable. 

After tinkering for a bit I think I'll just leave it as is.